packages/components/src/Tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,136 @@
+
+import { TooltipContext } from "./TooltipContext.ts";
+import { TooltipTriggerContext } from "./TooltipTriggerContext.ts";
+
+
+import styles from "./Tooltip.module.css";
+
+export const GlobalTooltipCssSelector = "hop-Tooltip";
+
+type PropsToOmit = "triggerRef" | "UNSTABLE_portalContainer" | "placement" | "containerPadding" | "offset" | "crossOffset" |
+"shouldFlip" | "arrowBoundaryOffset" | "isOpen" | "defaultOpen" | "onOpenChange";
+
+export type TooltipContainerProps = Omit<BaseComponentDOMProps, "slot"> & StyledSystemProps;
+
+export interface TooltipProps extends StyledComponentProps<Omit<RACTooltipProps, PropsToOmit>> {
+    /**
+     * The props of the tooltip's inner container.
+     */
+    containerProps?: TooltipContainerProps;
+}
+
+function Tooltip(props: TooltipProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, TooltipContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        children: childrenProp,
+        className,
+        containerProps,
+        style: styleProp,
+        ...otherProps
+    } = ownProps;
+
+    const {
+        containerPadding,
+        crossOffset,
+        offset,
+        placement = "top",
+        shouldFlip
+    } = useContext(TooltipTriggerContext);
+    
+    const { colorScheme } = useColorSchemeContext();
+    
+    const { stylingProps: containerStylingProps, ...containerOwnProps } = useStyledSystem(containerProps ?? {});
+    const {
+        className: containerClassName,
+        style: containerStyleProp,
+        ...containerOtherProps
+    } = containerOwnProps;
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalTooltipCssSelector,
+        cssModule(
+            styles,
+            "hop-Tooltip",
+            placement
+        ),
+        stylingProps.className
+    );
+
+    const containerClassNames = clsx(containerClassName, styles["hop-Tooltip__container"], containerStylingProps.className);
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const containerStyle = {
+        ...containerStylingProps.style,
+        ...containerStyleProp
+    };
+
+    const children = composeRenderProps(childrenProp, prev => {
+        return ensureTextWrapper(prev);
+    });
+
+    return (
+        <RACTooltip
+            ref={ref}
+            className={classNames}
+            style={style}
+            containerPadding={containerPadding}
+            crossOffset={crossOffset}
+            offset={offset}
+            placement={placement}
+            shouldFlip={shouldFlip}
+            {...otherProps}
+        >
+            {tooltipRenderProps => {
+                return (
+                    <HopperProvider colorScheme={colorScheme} withCssVariables={false}>
```

Comment:
[Nit] Not you, but I thought we wanted to avoid boolean that defaulted to true, is this a case where it's ok?

Reply:
I guess so? haha good point. might want to ask @alexasselin008 about that one.

Reply:
and the question for wrapping HopperProvider by Tooltip, did you mean put it around the tooltip? I'm guessing it wouldn't since the Tooltip would still be portaled outside of it.

Reply:
Think you're right